pkg/pkgrepository/app_fetch.go

```diff
@@ -55,12 +54,16 @@ func (a *App) fetch(dstPath string) (string, exec.CmdRunResult) {
 			// Sleep for 2 seconds to allow secretgen-controller
```

Comment:
```
		time.Sleep(2 * time.Second)
		newConf, err := vendir.ConfigBytes()
		if err != nil {
			result.AttachErrorf("Fetching: %v", err)
			return "", result
		}

		for i := 0; i < 2 && !bytes.Equal(conf, newConf); i++ {
			time.Sleep(2 * time.Second)
			newConf, err := vendir.ConfigBytes()
			if err != nil {
				result.AttachErrorf("Fetching: %v", err)
				return "", result
			}
			
			result = a.runVendir(newConf, dstPath)
			if result.Error == nil {
				break
			}
		}
```

How do you feel about this? I think the `continue` on line 64 will add 4 seconds of sleeping (by looping another 2 times potentially) that don't need to happen. Right?

Reply:
> on line 64 will add 4 seconds of sleeping (by looping another 2 times potentially) that don't need to happen. Right?

i think we do want that so that there is enough time for any change to kick to the secret. this would only happen if fetch failed in a first place.

Reply:
The idea is that after the next sleep `vendir.ConfigBytes()` might have returned a new Vendir configuration (with image pull secrets).

The flow is basically:
- resolve all the `secretRef`/`configMapRef`s and turn them into the [multi-doc config format](https://github.com/vmware-tanzu/carvel-vendir/blob/develop/examples/secrets/vendir.yml) that vendir supports 
- run `vendir sync`
- if it fails and one of our fetches is an `image:` or `imgpkg:`
  - resolve all the secrets/configmaps again
  - if they're the same, sleep/retry
  - if they're different run `vendir sync` again
  - if it fails again, sleep/retry

Reply:
> The idea is that after the next sleep vendir.ConfigBytes() might have returned a new Vendir configuration (with image pull secrets).

gotcha! Coolio. Can disregard the comment so 👍